addons/stock/models/stock_move.py

```diff
@@ -598,6 +598,13 @@ def write(self, vals):
                 receipt_moves_to_reassign |= move_to_unreserve.filtered(lambda m: m.location_id.usage == 'supplier')
                 receipt_moves_to_reassign |= (self - move_to_unreserve).filtered(lambda m: m.location_id.usage == 'supplier' and m.state in ('partially_available', 'assigned'))
                 move_to_recompute_state |= self - move_to_unreserve - receipt_moves_to_reassign
+        # propagate packaging_id changes in the stock move chain
+        if 'product_packaging_id' in vals:
+            if self.move_dest_ids and \
+                    self.move_dest_ids.state not in ['cancel', 'done'] and \
+                    self.move_dest_ids.product_packaging_id.id != vals['product_packaging_id'] and \
+                    self.move_dest_ids.move_orig_ids == self:  # checks that you are the only parent move of your destination
```

Comment:
I'm not sure of that one. Suppose `self` contains SM01 and SM02
SM01 has a dest move, and the origin of the dest is SM01 (-> ok)
SM02 does not have any dest move
We will have `self.move_dest_ids.move_orig_ids == SM01 | SM02 != self` so we will not enter in the if-condition, but we should since SM01.dest needs to be updated
I think this could also be part of the test